commands/utils.go

```diff
@@ -195,13 +203,97 @@ func buildReport(results []Result) error {
 			io.Copy(os.Stdout, rs.Output.Stderr())
 
 			fmt.Println()
+			timeElapsedArray[i] = rs.TimeElapsed
+			nodeIDArray[i] = rs.Node
 		}
 
 	}
+	if statsFlag {
+		stats, err := BuildStats(timeElapsedArray)
+		if err != nil {
+			errs = append(errs, err)
+		} else {
+			statsJSON, _ := json.Marshal(stats)
+			fmt.Printf("Executed command < %s > on %d node(s) \nTime Statistics: %s \n", command, len(results), string(statsJSON))
+			printIndividualResults(timeElapsedArray, nodeIDArray, "time")
+		}
+	}
 
 	if len(errs) != 0 {
 		return cli.NewMultiError(errs...)
 	}
 
 	return nil
 }
+
+func buildMetricStats(metricsBefore []string, metricsAfter []string, metric string) error {
+	var errs []error
+
+	// Substract Before from After to generate the results
+	results, err := subtractArrays(metricsBefore, metricsAfter)
+	if err != nil {
+		return err
+	}
+	// Create an array of node IDs
+	// TODO: Make this pretty :), and eliminate this for-loop
+	nodesID := make([]int, len(metricsAfter))
+
+	for i := range metricsAfter {
+		nodesID[i] = i
+	}
+	// Use the build stats function to generate the stats
+	stats, err := BuildStats(results)
+	if err != nil {
+		errs = append(errs, err)
+	} else {
+		statsJSON, _ := json.Marshal(stats)
+		fmt.Printf("Metric collected < %s > on %d node(s) \nStatistics: %s \n", metric, len(metricsAfter), string(statsJSON))
+		printIndividualResults(results, nodesID, metric)
+	}
+	if len(errs) != 0 {
+		return cli.NewMultiError(errs...)
+	}
+	return nil
+}
+
+// Print individual results to JSON format
+func printIndividualResults(nodeTime []float64, nodeID []int, metric string) {
+	fmt.Printf("Analytic Results: {[")
+	for i, time := range nodeTime {
+		fmt.Printf("{Node: %d, %s: %f}", nodeID[i], metric, time)
+		if i != len(nodeTime)-1 {
+			fmt.Printf(",")
+		}
+	}
+	fmt.Printf("]},")
+}
+
+/*
+Substract_arrays
+Substract y from x returns [xi-yi] for i in len(x)
+If the arrays do not have the same length or
+the strings are not castable to float
+the function returns an error
+TODO: This could be more generic
+*/
+func subtractArrays(lhs []string, rhs []string) ([]float64, error) {
```

Comment:
As it is implemented now the metrics are collected as follows:
Let metric <m1> be collected, then we collect m1<sub>before</sub> before the execution of the run command and m1<sub>after</sub> after. Then we calculate the statistics based on Δm1 = (m1<sub>after</sub> - m1<sub>before</sub>).